### PR TITLE
Adding AdvPromptSet to ResponsibleNLP repo

### PR DESCRIPTION
**Pull request by ellenxtan**
Please refer to `README.md` for instruction to generate the AdvPromptSet dataset.

### CODE REVIEW

File: AdvPromptSet/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+### Generating the dataset
+
+Create a folder `AdvPromptSet/out_data` to store the generated dataset. Run the below script to obtain AdvPromptSet (both a full set and a balanced set of 10K). The output files are in `out_data` folder.
+
+```
+mkdir AdvPromptSet/out_data
+python AdvPromptSet/main.py
+```
+
+The final structure of the directory is shown below.
```
Comment (EricMichaelSmith) [inline AdvPromptSet/README.md:55]: Love this detail, it's great to have these checks!!

File: AdvPromptSet/main.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import os
+import pandas as pd
+from utils import process_jigsaw_1
+from utils import process_jigsaw_2
+from utils import process_jigsaw_all
+from utils import check_shape
+
+
+if __name__ == "__main__":
```
Comment (EricMichaelSmith) [inline AdvPromptSet/main.py:10]: Maybe give a docstring or just a comment at the top saying what this script does? People might see this script without seeing the README first

File: AdvPromptSet/utils.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
+import os
+import numpy as np
+import pandas as pd
+
+
+def process_jigsaw_1(dat_folder):
```
Comment (EricMichaelSmith) [inline AdvPromptSet/utils.py:6]: could be good for future reference to add docstrings saying quickly what each of these do in a nutshell

File: AdvPromptSet/utils.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        df = remove_long_rows(df)
+        df = augment_row(df)
+        df = df[["id", "comment_text", "toxicity", "jigsaw_dat", "test_set", "identity_grp"]].copy()
+        df_lst_processed.append(df)
+
+    df_jigsaw = pd.concat(df_lst_processed, axis=0).reset_index(drop=True).copy()
+    return df_jigsaw
+
+
+def check_shape(term, df):
```
Comment (EricMichaelSmith) [inline AdvPromptSet/utils.py:104]: Hmm - what is the output supposed to be here? One idea could be to add an `assert` to make sure it's the correct shape/counts, up to you

File: AdvPromptSet/utils.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    print(term, df.shape)
+    print(df['toxicity'].value_counts())
+
+
+def tokenized_words(sentence):
+    from nltk.tokenize import RegexpTokenizer
```
Comment (EricMichaelSmith) [inline AdvPromptSet/utils.py:110]: maybe cleaner to put this import at the top?

**Review by EricMichaelSmith — COMMENTED** [review]
This is great to have! Might be better to pull these large metadata files from the cloud vs. add them here into the PR, so that folks using FairScore or HolisticBias code don't have to download this data as part of `git clone` - will ping about that separately

File: AdvPromptSet/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
+## AdvPromptSet
```
[the remark this answers is not in the document]
Comment (ellenxtan) [reply AdvPromptSet/README.md:1]: Added! Let's revisit the format later to coordinate with other open sourcing stuffs.

File: AdvPromptSet/main.py (hunk 2 above)
Comment (ellenxtan) [reply AdvPromptSet/main.py:15]: Make sense! added

File: AdvPromptSet/utils.py (hunk 3 above)
Comment (ellenxtan) [reply AdvPromptSet/utils.py:7]: Make sense! added docstrings for all functions

File: AdvPromptSet/utils.py (hunk 5 above)
Comment (ellenxtan) [reply AdvPromptSet/utils.py]: Make sense! Moved the import to the top and add the related packages in `requirements.txt`.

File: AdvPromptSet/utils.py (hunk 4 above)
Comment (ellenxtan) [reply AdvPromptSet/utils.py]: Removed. Thanks for catching. Those are scripts for debugging purposes.

**Review by EricMichaelSmith — APPROVED** [review]
Revisions look great, thanks!

**Pull request merged**